pkgs/development/php-packages/datadog_trace/default.nix

```
@@ -1,22 +1,51 @@
-{ buildPecl, curl, fetchFromGitHub, lib, pcre2, php }:
+{ lib
+, buildPecl
+, stdenv
+, fetchFromGitHub
+, rustPlatform
+, darwin
+, curl
+, pcre2
+}:
 
 buildPecl rec {
-  pname = "ddtrace";
-  version = "0.82.0";
+  pname = "datadog_trace";
+  version = "0.89.0";
 
   src = fetchFromGitHub {
     owner = "DataDog";
     repo = "dd-trace-php";
     rev = version;
-    sha256 = "sha256-QTqZRHh57mRkg0HT9qQS13emGobB0IRqM+mdImAPgtE=";
+    hash = "sha256-wTGQV80XQsBdmTQ+xaBKtFwLO3S+//9Yli9aReXDlLA=";
+    fetchSubmodules = true;
   };
 
-  buildInputs = [ curl pcre2 ];
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "datadog-profiling-2.2.0" = "sha256-PWzC+E2u0hM0HhU0mgZJZvFomEJdQag/3ZK1FibSLG8=";
+    };
+  };
+
+  env.NIX_CFLAGS_COMPILE = "-O2";
+
+  nativeBuildInputs = [
+    rustPlatform.bindgenHook
+    rustPlatform.cargoSetupHook
+    rustPlatform.rust.cargo
+    rustPlatform.rust.rustc
```

`rustPlatform.rust.*` has been deprecated, this has been fixed in nix-init 0.2.4 / nixpkgs master

```suggestion
    cargo
    rustc
```

---

Fixed.